src/main/java/com/google/api/codegen/discovery/transformer/go/GoSampleMethodToViewTransformer.java

```diff
@@ -142,9 +147,15 @@ private SampleAuthView createSampleAuthView(SampleTransformerContext context) {
     MethodInfo methodInfo = config.methods().get(context.getMethodName());
 
     List<String> scopeConsts = new ArrayList<>();
-    // Pull the last scope and reconstruct it as a Go constant.
+    // Pull scopes and reconstruct them as Go constants.
     // For example: "https://www.googleapis.com/auth/cloud-platform" to "CloudPlatform"
-    if (methodInfo.authScopes().size() > 0) {
+    // If the sample uses ADC auth, only add the first scope. Otherwise, add all
```

Comment:
First scope is always CloudPlatform, though I was debating just enforcing this in the snippet instead.